coalib/core/Core.py

```diff
@@ -327,34 +147,172 @@ def instantiate_and_track(prev_bear_type, next_bear_type):
     return dependency_tracker, bears
 
 
-def run(bears, result_callback):
-    """
-    Runs a coala session.
-
-    :param bears:
-        The bear instances to run.
-    :param result_callback:
-        A callback function which is called when results are available. Must
-        have following signature::
-
-            def result_callback(result):
-                pass
-    """
-    # FIXME Allow to pass different executors nicely, for example to execute
-    # FIXME   coala with less cores, or to schedule jobs on distributed systems
-    # FIXME   (for example Mesos).
-
-    # Set up event loop and executor.
-    event_loop = asyncio.SelectorEventLoop()
-    executor = concurrent.futures.ProcessPoolExecutor()
-
-    # Initialize dependency tracking.
-    dependency_tracker, bears_to_schedule = initialize_dependencies(bears)
-
-    # Let's go.
-    schedule_bears(bears_to_schedule, result_callback, dependency_tracker,
-                   event_loop, {}, executor)
-    try:
-        event_loop.run_forever()
-    finally:
-        event_loop.close()
+class Session:
+
+    @classmethod
+    def run(cls, bears, result_callback):
+        """
+            Runs a coala session.
+
+            :param bears:
+                The bear instances to run.
+            :param result_callback:
+                A callback function which is called when results are available.
+                Must have following signature::
+
+                    def result_callback(result):
+                        pass
+            """
+        cls.bears = bears
```

Comment:
ahh no make it instance bound, that is way safer and then we can execute multiple sessions at once :D